TestPlaybooks/playbook-TestExtractHTMLTables.yml

```diff
@@ -0,0 +1,445 @@
+version: -1
+name: TestExtractHTMLTables
+starttaskid: "0"
+tasks:
+  "0":
+    id: "0"
+    taskid: 27097a72-8399-4e27-85f5-897c27106ce3
+    type: start
+    task:
+      id: 27097a72-8399-4e27-85f5-897c27106ce3
+      version: -1
+      name: ""
+      iscommand: false
+      brand: ""
+    nexttasks:
+      '#none#':
+      - "1"
+    separatecontext: false
+    view: |-
+      {
+        "position": {
+          "x": 245,
+          "y": 50
+        }
+      }
+  "1":
+    id: "1"
+    taskid: 4447dd62-1e51-483b-802f-202205b781f4
+    type: regular
+    task:
+      id: 4447dd62-1e51-483b-802f-202205b781f4
+      version: -1
+      name: DeleteContext
+      description: Delete field from context
+      scriptName: DeleteContext
+      type: regular
+      iscommand: false
+      brand: ""
+    nexttasks:
+      '#none#':
+      - "2"
+    scriptarguments:
+      all:
+        simple: "yes"
+      key: {}
+    separatecontext: false
+    view: |-
+      {
+        "position": {
+          "x": 245,
+          "y": 195
+        }
+      }
+  "2":
+    id: "2"
+    taskid: 6cc1640f-24df-4b6a-8cf2-5d2daaac5c57
+    type: regular
+    task:
+      id: 6cc1640f-24df-4b6a-8cf2-5d2daaac5c57
+      version: -1
+      name: ExtractHTMLTables with single one column table with header
+      description: |-
+        Find tables inside HTML and extract the contents into objects using the following logic:
+        - If table has a single column, just create an array of strings from the values
+        - If table has 2 columns and has no header row, treat the first column as key and second as value and create a table of key/value
+        - If table has a header row, create a table of objects where attribute names are the headers
+        - If table does not have a header row, create table of objects where attribute names are cell1, cell2, cell3...
+      scriptName: ExtractHTMLTables
+      type: regular
+      iscommand: false
+      brand: ""
+    nexttasks:
+      '#none#':
+      - "3"
+    scriptarguments:
+      html:
+        simple: <html><body><table><tr><th>kuku</th></tr><tr><td>a</td></tr><tr><td>1</td></tr></table></body></html>
+      indexes: {}
+    separatecontext: false
+    view: |-
+      {
+        "position": {
+          "x": 245,
+          "y": 370
+        }
+      }
+  "3":
+    id: "3"
+    taskid: 9f902e22-b34e-4559-8238-8752c1afda3e
+    type: condition
+    task:
+      id: 9f902e22-b34e-4559-8238-8752c1afda3e
+      version: -1
+      name: Both values exist in array
+      description: Check if a given value exists in the context. Will return 'no'
+        for empty empty arrays. To be used mostly with DQ and selectors.
+      scriptName: Exists
+      type: condition
+      iscommand: false
+      brand: ""
+    nexttasks:
+      '#default#':
+      - "5"
+      "yes":
+      - "6"
+    scriptarguments:
+      value:
+        simple: '${HTMLTables=>val[0] == ''a'' && val[1] == ''1'' ? ''x'' : ''''}'
```

Comment:
`'${HTMLTables=>val[0] == ''a'' && val[1] == ''1'' ? ''x'' : ''''}'` should be `'${HTMLTables=val[0] == ''a'' && val[1] == ''1'' ? ''x'' : ''''}'`
replace `=>` to `=` in all the playbook

Reply:
Why??? `=>` is the correct form in this case because I want the array

Reply:
You right